Pod/Classes/public/error/Error.swift

```diff
@@ -0,0 +1,150 @@
+//
+//  Error.swift
+//  Pods
+//
+//  Created by Jake Young on 6/14/16.
+//
+
+import Foundation
+
+/// Debugging levels for WistiaKit
+public enum DebuggingLevel {
+    
+    /// Shhh, fust be silent.
+    case None
+    
+    /// Print the high level activity and errors.
+    case Friendly
+    
+    /// Print all the things.
+    case Annoying
+}
+
+internal func printErrorForDebuggingLevel(debuggingLevel debuggingLevel: DebuggingLevel, message: String? = nil, error: NSError? = nil) {
+    
+    switch debuggingLevel {
+        
+    case .None:
+        break
+    case .Friendly:
+        
+        // We print the message and fallthrough to allow .Annoying case to add to things.
+        guard let message = message else { break }
+        print(message)
+        fallthrough
+        
+    case .Annoying:
+        
+        // We unrap the provided erorr and print the nerdy details.
+        guard let error = error else { break }
+        print("Error: \(error.description)), reason: \(error.localizedFailureReason)")
+    }
+    
+}
+
+
+/// The `Error` struct provides a convenience for creating custom WistiaKit NSErrors and Swift ErrorTypes.
+public struct Error {
+    
+    /// The domain used for creating all Wistia errors.
+    public static let Domain = "com.wistia.error"
```

Comment:
Apple is pushing Swift hard.  It's the future, if not the present.  I'm leaning heavily in favor of following that pattern unless/until there is a very compelling reason to increase ObjC interop.

Dropping NSError (and its related additions) makes this simpler and clearer.